ironic/dualboot.ipxe

```diff
@@ -0,0 +1,28 @@
+#!ipxe
+
+# NOTE(lucasagomes): Loop over all network devices and boot from
+# the first one capable of booting. For more information see:
+# https://bugs.launchpad.net/ironic/+bug/1504482
+set netid:int32 -1
+:loop
+inc netid || chain pxelinux.cfg/${mac:hexhyp} || goto old_rom
+isset ${net${netid}/mac} || goto inspector 
```

Comment:
I'm not sure what this does. I'm quite confident this `isset` will hold for most sane hardware.

Reply:
To me this looks like execution keeps going to "loop:", incrementing  netid each time until this line fails (i.e. no more nics left to try), then it goes to inspector, without this line we're stuck in the loop?

Reply:
Right, I think this should be
```
isset ${net${netid}/mac} || chain pxelinux.cfg/${mac:hexhyp} || goto inspector
```
(to keep fallback to `mac:hexhyp`, then fallback to inspector).

Reply:
That just happens on the next line, isn't it the same thing? 
> isset ${net${netid}/mac} || goto inspector 
> chain pxelinux.cfg/${net${netid}/mac:hexhyp} || goto loop 

if ${net${netid}/mac} is not set goto inspector
else chain....

Reply:
No, because you need to also try `${mac:hexhyp}` in case of really old iPXE ROM (at least that's what ironic does).

Reply:
I don't see where ironic does that, all I see is
http://git.openstack.org/cgit/openstack/ironic/tree/ironic/drivers/modules/boot.ipxe#n9
> isset ${net${netid}/mac} || goto loop_done

Reply:
Yes ironic doesn't include the inspector support, just terminates if can't find the find in pxelinux.cfg